### PR TITLE
[e2e tests]: Use new MatchError form instead of boolean IsNotFound assertions

### PR DESCRIPTION
**Pull request by akalenyu**
**What this PR does / why we need it**:
Using the new [form](https://github.com/onsi/gomega/commit/2b39142b16e69fa9405d5c7401dc62a5776a9bf5) of this matcher will provide more context in test failure error messages,
and should overall make the assertions nicer to read.

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #

**Special notes for your reviewer**:
This is how failure messages for these assertions look like today:
image

**Checklist**

This checklist is not enforcing, but it's a reminder of items that could be relevant to every PR.
Approvers are expected to review this list.

- [ ] Design: A [design document](https://github.com/kubevirt/community/tree/main/design-proposals) was considered and is present (link) or not required
- [ ] PR: The PR description is expressive enough and will help future contributors
- [ ] Code: [Write code that humans can understand](https://en.wikiquote.org/wiki/Martin_Fowler#code-for-humans) and [Keep it simple](https://en.wikipedia.org/wiki/KISS_principle)
- [ ] Refactor: You have [left the code cleaner than you found it (Boy Scout Rule)](https://learning.oreilly.com/library/view/97-things-every/9780596809515/ch08.html)
- [ ] Upgrade: Impact of this change on upgrade flows was considered and addressed if required
- [ ] Testing: New code requires [new unit tests](https://github.com/kubevirt/kubevirt/blob/main/docs/reviewer-guide.md#when-is-a-pr-good-enough). New features and bug fixes require at least on e2e test
- [ ] Documentation: A [user-guide update](https://github.com/kubevirt/user-guide/) was considered and is present (link) or not required. You want a user-guide update if it's a user facing feature / API change.
- [ ] Community: Announcement to [kubevirt-dev](https://groups.google.com/g/kubevirt-dev/) was considered

**Release note**:

```release-note
NONE
```

### CODE REVIEW

**Review by EdDev — APPROVED** [review]
Thank you!

File: tests/libwait/wait.go [hunk 1]
```diff
@@ -211,8 +211,8 @@ func WaitForVirtualMachineToDisappearWithTimeout(vmi *v1.VirtualMachineInstance,
 func WaitForMigrationToDisappearWithTimeout(migration *v1.VirtualMachineInstanceMigration, seconds int) {
 	virtClient, err := kubecli.GetKubevirtClient()
 	gomega.ExpectWithOffset(1, err).ToNot(gomega.HaveOccurred())
-	gomega.EventuallyWithOffset(1, func() bool {
+	gomega.EventuallyWithOffset(1, func() error {
 		_, err := virtClient.VirtualMachineInstanceMigration(migration.Namespace).Get(migration.Name, &metav1.GetOptions{})
-		return errors.IsNotFound(err)
-	}, seconds, 1*time.Second).Should(gomega.BeTrue(), fmt.Sprintf("migration %s was expected to dissapear after %d seconds, but it did not", migration.Name, seconds))
+		return err
+	}, seconds, 1*time.Second).Should(gomega.MatchError(errors.IsNotFound, "k8serrors.IsNotFound"), fmt.Sprintf("migration %s was expected to dissapear after %d seconds, but it did not", migration.Name, seconds))
```
Comment (codingben) [inline tests/libwait/wait.go:217]: `disappear`

File: tests/vmipreset_test.go [hunk 2]
```diff
@@ -577,14 +577,8 @@ func getPreset(virtClient kubecli.KubevirtClient, prefix string) (*v1.VirtualMac
...
-		if err != nil {
-			if !errors.IsNotFound(err) {
-				Expect(err).ToNot(HaveOccurred())
-			}
-			return true
-		}
-		return false
-	}, time.Duration(60)*time.Second).Should(BeTrue(), "timed out waiting for preset to be deleted")
+		return err
+	}, 60*time.Second, 1*time.Second).Should(MatchError(errors.IsNotFound, "k8serrors.IsNotFound"), "timed out waiting for preset to be deleted")
```
Comment (codingben) [inline tests/vmipreset_test.go:583]: `timed out waiting for VMI preset to be deleted`

File: tests/vmi_lifecycle_test.go [hunk 3]
```diff
@@ -711,10 +711,10 @@ var _ = Describe("[rfe_id:273][crit:high][arm64][vendor:cnv-qe@redhat.com][level
 				_, err = kubevirt.Client().KubeVirt(kv.Namespace).Update(kv)
 				Expect(err).ToNot(HaveOccurred(), "Should update kubevirt infra placement")
 
-				Eventually(func() bool {
+				Eventually(func() error {
 					_, err := kubevirt.Client().CoreV1().Pods(virtHandler.Namespace).Get(context.Background(), virtHandler.Name, metav1.GetOptions{})
-					return k8serrors.IsNotFound(err)
-				}, 120*time.Second, 1*time.Second).Should(BeTrue(), "The virthandler pod should be gone")
+					return err
+				}, 120*time.Second, 1*time.Second).Should(MatchError(k8serrors.IsNotFound, "k8serrors.IsNotFound"), "The virthandler pod should be gone")
```
Comment (codingben) [inline tests/vmi_lifecycle_test.go:717]: `virt-handler`, if you'd like to change it

**Review by codingben — APPROVED** [review]
/lgtm

File: tests/operator/operator.go [hunk 4]
```diff
@@ -1851,19 +1851,10 @@ spec:
...
-						Expect(err).ToNot(HaveOccurred())
-					}
-					return false
-					// #3610 - this timeout needs to be reduced back to 60 seconds.
-					// there's an issue occurring after update where sometimes virt-launcher
-					// can't dial the event notify socket. This impacts the timing for when
-					// the vmi is shutdown. Once that is resolved, reduce the timeout
-				}, 160*time.Second, 1*time.Second).Should(BeTrue())
+					return err
+				}, 60*time.Second, 1*time.Second).Should(MatchError(errors.IsNotFound, "k8serrors.IsNotFound"))
```
Comment (AlonaKaplan) [inline tests/operator/operator.go:1857]: The reduction to 60 seconds deserves a separate commit and a link to the resolution.

File: tests/pausing_test.go [hunk 5]
```diff
@@ -285,14 +285,10 @@ var _ = Describe("[rfe_id:3064][crit:medium][vendor:cnv-qe@redhat.com][level:com
...
+				Eventually(func() error {
 					_, err = virtClient.VirtualMachineInstance(vm.Namespace).Get(context.Background(), vm.Name, &v12.GetOptions{})
-					if errors.IsNotFound(err) {
-						return true
-					}
-					Expect(err).ToNot(HaveOccurred())
-					return false
-				}, 300*time.Second, 1*time.Second).Should(BeTrue(), "The VMI did not disappear")
+					return err
+				}, 300*time.Second, 1*time.Second).Should(MatchError(errors.IsNotFound, "k8serrors.IsNotFound"))
```
Comment (AlonaKaplan) [inline tests/pausing_test.go:291]: I find the previous "The VMI did not disappear" description useful.

File: tests/operator/operator.go (hunk 4 above)
Comment (akalenyu) [reply tests/operator/operator.go:1857]: done, ty

File: tests/pausing_test.go (hunk 5 above)
Comment (akalenyu) [reply tests/pausing_test.go]: yup whoops, readded

File: tests/vmipreset_test.go (hunk 2 above)
Comment (akalenyu) [reply tests/vmipreset_test.go]: done

File: tests/libwait/wait.go (hunk 1 above)
Comment (akalenyu) [reply tests/libwait/wait.go]: thx

**Review by EdDev — APPROVED** [review]
Thanks!

It would have been nice to provide a context topic (e.g. "tests: ") to all commits subjects, but I will not insist on that now.

File: tests/operator/operator.go [hunk 6]
```diff
@@ -1851,19 +1851,10 @@ spec:
 				}, 30*time.Second, 1*time.Second).Should(BeNil())
 
 				By("Waiting for VMI to stop")
-				Eventually(func() bool {
+				Eventually(func() error {
```
Comment (AlonaKaplan) [inline tests/operator/operator.go:1854]: >  Seems like this was fixed somewhere between https://github.com/kubevirt/kubevirt/issues/3610 and now.

How do you know it was fixed? The fact if passed several times with 60 sec doesn't mean the issue was fixed. AFAIU, the test was flaky and didn't always fail.
But you're right, it is old, we can give it a chance. I"m approving, let's see if we see the flake again.

**Comment by oshoval** [comment]
nit: maybe worth to add to the PR desc how it will look with the new style

File: tests/operator/operator.go (hunk 6 above)
Comment (akalenyu) [reply tests/operator/operator.go:1854]: Yeah we don't know for sure it's fixed but we should be concerned if it's still needed
since currently it's just hiding potential issues from us
